clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -45,7 +45,7 @@
  * A connection can be added to the nioSelector associated with an integer id by doing
  * 
  * <pre>
- * nioSelector.connect(42, new InetSocketAddress(&quot;google.com&quot;, server.port), 64000, 64000);
+ * nioSelector.connect(&quot;42&quot;, new InetSocketAddress(&quot;google.com&quot;, server.port), 64000, 64000);
```

Comment:
Thanks for fixing. I changed the API and forgot the doc. Selector used to take a broker-id as identifier when it was used just for client-broker communications. Once we needed it for inter-broker, we needed more flexibility and changed that to String.